packet/flatcar-linux/kubernetes/workers/cl/install.yaml.tmpl

```diff
@@ -88,14 +88,57 @@ storage:
             mkfs.ext4 /dev/md/node-local-storage
           }
 
+          function create_disk_specific_data_raid() {
+            # Ignore the disk on which Linux is installed when selecting disks for RAID.
+            local osdisk="$1"
+            local major_numbers="$2"
+            # for hdd value is 1 and for ssd value is 0
+            local disk_type="$3"
+            # RAID device path on disk
+            local device_path="$4"
+            local setup_fs_on_raid="$5"
+
+            # Select disks for RAID.
+            local disks=$(lsblk -lnpd -o name,rota -I "$${major_numbers}" \
+              | grep "$${disk_type}" \
+              | sort -h -k 4,4 \
```

Comment:
This `sort` can be deleted and has no effect anyway because the keys are wrong. If you want to sort by something, then use `lsblk -lnpd - x size` or `lsblk -lnpd -x rota` (and add `| tac` to reverse the ordering).

Reply:
TIL: `tac` :smile:

Reply:
Now that I think about it, is there really a need to sort here? Do we need sorted disk order to form RAID?

Reply:
I think your suggestion should be added in https://github.com/kinvolk/lokomotive-kubernetes/blob/85732d1a934ddcd83fa8ea0a566f1ab3f89d0f66/packet/flatcar-linux/kubernetes/workers/cl/install.yaml.tmpl#L54-L58

Reply:
I think it would be good to have predictable RAID devices here, so sorting is a good idea. Otherwise for every server, you will have a different disk layout. It's easier to spot potential inconsistencies/issues when the setup is predictable.

Reply:
I think we should change all that in a follow up PR, this PR has already been prolonged for so long on very minuscule issues.

Reply:
There is no need to sort it I would say. The sorting was just done to find the smallest disk. Anyway, the `sort` invocation here is a no-op and should be deleted.

Reply:
For the record: `lsblk -lnpd -o name,rota  | sort -h -k 4,4 --debug` complains that it cannot match the key (sorting by `rota` makes no sense anyway?).

Reply:
Yeah it has to be sorted on the basis of size.

The diff would look something like this:

```diff
diff --git a/packet/flatcar-linux/kubernetes/workers/cl/install.yaml.tmpl b/packet/flatcar-linux/kubernetes/workers/cl/install.yaml.tmpl                                     
index f51c6df..39b63f2 100644                                                                                                                                                
--- a/packet/flatcar-linux/kubernetes/workers/cl/install.yaml.tmpl                                                                                                           
+++ b/packet/flatcar-linux/kubernetes/workers/cl/install.yaml.tmpl                                                                                                           
@@ -99,9 +99,8 @@ storage:                                                                                                                                                   
             local setup_fs_on_raid="$5"                                                                                                                                     
                                                                                                                                                                             
             # Select disks for RAID.                                                 
-            local disks=$(lsblk -lnpd -o name,rota -I "$${major_numbers}" \          
+            local disks=$(lsblk -lnpd -o name,rota,size -I "$${major_numbers}" -x size \                                                                                    
               | grep "$${disk_type}" \                                               
-              | sort -h -k 4,4 \                                                     
               | grep -vE "^$${osdisk} " \                                                                                                                                   
               | awk '{x=$1 " " x;} END{print x}'                                     
             ) 
```